### ISSUE TITLE
Modifying PWD should issue a warning, since it creates undefined behavior for "cd"

### ISSUE THREAD

**Issue by ozkar99** [koalaman/shellcheck#2160]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): 0.7.1
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

According to https://pubs.opengroup.org/onlinepubs/9699919799/utilities/cd.html on "ENVIRONMENT VARIABLES"
section:

```
PWD
This variable shall be set as specified in the DESCRIPTION. If an application sets or unsets the value of PWD, the behavior of cd is unspecified.
```

Not sure if this is a reasonable thing to lint/watch out for, as there might be a lot more "unsafe" env vars in other use cases, but since I didn't saw a precedent for this I am bringing it up (up to shellcheck maintainers to deem this as a non-feature).

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh

PWD=""

```

#### Here's what shellcheck currently says:
No issues detected!

#### Here's what I wanted or expected to see:
Warning: Modifying PWD can cause undefined behavior for `cd`.
